ocs_ci/ocs/cluster.py

```diff
@@ -3790,3 +3792,101 @@ def get_age_of_cluster_in_days():
     seconds_per_day = 24 * 60 * 60
     time_diff_in_days = time_difference_in_sec / seconds_per_day
     return math.ceil(time_diff_in_days)
+
+
+def get_active_mds_count():
+    """
+    Get the active mds count from the system.
+
+    Returns:
+         int: active_mds_count
+
+    """
+    cephfs = ocp.OCP(
+        kind=constants.CEPHFILESYSTEM,
+        namespace=config.ENV_DATA["cluster_namespace"],
+    )
+    fs_data = cephfs.get(defaults.CEPHFILESYSTEM_NAME)
+    mds_active_count = fs_data.get("spec").get("metadataServer").get("activeCount")
+    return mds_active_count
+
+
+def adjust_active_mds_count(target_count):
+    """
+    Adjust the activeMetadataServers count for the Storage cluster to the target_count.
+    The function will increase or decrease the count to match the target value.
+
+    Args:
+        target_count (int): The desired count for activeMetadataServers.
+
+    """
+
+    # Retrieve the current activeMetadataServers count
+    current_count = get_active_mds_count()
+    sc = get_storage_cluster(namespace=config.ENV_DATA["cluster_namespace"])
+    resource_name = sc.get()["items"][0]["metadata"]["name"]
+
+    if current_count < target_count:
+        # Increment mds pods
+        while current_count < target_count:
+            current_count += 1
+            param = (
+                f'{{"spec": {{"managedResources": {{"cephFilesystems": '
+                f'{{"activeMetadataServers": {current_count}}}}}}}}}'
+            )
+            sc.patch(resource_name=resource_name, params=param, format_type="merge")
+            current_params = sc.get(resource_name=resource_name)
+            current_count = current_params["spec"]["managedResources"][
+                "cephFilesystems"
+            ]["activeMetadataServers"]
+
+    elif current_count > target_count:
+        # Decrement mds pods
+        while current_count > target_count:
+            current_count -= 1
+            param = (
+                f'{{"spec": {{"managedResources": {{"cephFilesystems": '
+                f'{{"activeMetadataServers": {current_count}}}}}}}}}'
+            )
+            sc.patch(resource_name=resource_name, params=param, format_type="merge")
+            current_params = sc.get(resource_name=resource_name)
+            current_count = current_params["spec"]["managedResources"][
+                "cephFilesystems"
+            ]["activeMetadataServers"]
+
+    else:
+        logger.info(
+            "The current count is already equal to the target count. No changes needed."
+        )
+
+    # Retrieve the current activeMetadataServers count again
+    logger.info("Sleeping 1 minute")
+    time.sleep(60)
```

Comment:
can we avoid this hardcoded sleep replacing it with polling some resource status?